manager/orchestrator/global/global.go

```diff
@@ -142,16 +145,16 @@ func (g *Orchestrator) Run(ctx context.Context) error {
 				if !orchestrator.IsGlobalService(v.Service) {
 					continue
 				}
-				g.updateService(v.Service)
-				g.reconcileServices(ctx, []string{v.Service.ID})
-			case api.EventDeleteService:
-				if !orchestrator.IsGlobalService(v.Service) {
-					continue
+				if v.Service.PendingDelete {
+					// ask the tasks to shut down
+					orchestrator.SetServiceTasksRemove(ctx, g.store, v.Service)
+					// delete the service from service map
+					delete(g.globalServices, v.Service.ID)
```

Comment:
Once removed from here, is the service still accessible from UI/CLI using list commands etc? Is that the behavior we're going for?

Reply:
Yes it is. There will be a CLI change to explicitly mark it as pending for deletion in the CLI's output.